client/issue-lib.js

```diff
@@ -0,0 +1,144 @@
+(function() {
+'use strict';
+
+if (window.Issue) {
+  return;
+}
+
+// Colours are copied from:
+// https://www.google.com/design/spec/style/color.html#color-color-palette
+var _reviewLevelMetadata = {
+  'daily': {
+    query: {label: 'Update-Daily'},
+    outOfSLOColor: '#B71C1C',  // Red 900
+  },
+  'weekly': {
+    query: {label: 'Update-Weekly'},
+    outOfSLOColor: '#C62828',  // Red 800
+  },
+  'fortnightly': {
+    query: {label: 'Update-Fortnightly'},
+    outOfSLOColor: '#D32F2F',  // Red 700
+  },
+  'monthly': {
+    query: {label: 'Update-Monthly'},
+    outOfSLOColor: '#E53935',  // Red 600
+  },
+  'quarterly': {
+    query: {label: 'Update-Quarterly'},
+    outOfSLOColor: '#F44336',  // Red 500
+  },
+  'none (P0)': {
+    query: {label: 'Pri-0', '-has': 'update'},
+  },
+  'none (P1)': {
+    query: {label: 'Pri-1', '-has': 'update'},
+  },
+  'none (P2)': {
+    query: {label: 'Pri-2', '-has': 'update'},
+  },
+  'none (P3)': {
+    query: {label: 'Pri-3', '-has': 'update'},
+  },
+};
+var _defaultReviewLevel = 'none';
+var _inSLOColor = '#4CAF50';  // Green 500
+var _noSLOColor = '#757575';  // Grey 600
+
+// This class uses C style programming with JSON objects (representing issues)
+// and static methods.
+// This is to maintain the ability to clone the object in chromez-behaviours.js.
+class Issue {
+  static create({id, owner, summary, lastUpdatedString, labels}) {
+    var issue = {
+      id,
+      owner,
+      summary,
+      labels,
+      priority: undefined,
+      _lastUpdatedMS: Date.parse(lastUpdatedString),
```

Comment:
It just occurred to me: Can we rely on Monorail and GitHub both providing lastUpdatedString in the same format, or at least both in formats that Date.parse will accept? Might need to add a precondition for Issue.create that specifies the format of the arguments (mainly lastUpdatedString, but possibly also labels).